### ISSUE TITLE
distsql: trace info is not correctly propagated

### ISSUE THREAD

**Issue by andreimatei** [cockroachdb/cockroach#14816]
```
make testshort PKG=./pkg/sql/distsqlrun TESTFLAGS=-v TESTS=TestClusterFlow
```
prints in the log:
```
W170411 16:29:00.796719 536 sql/distsqlrun/server.go:96  [n3] failed to join a remote trace: opentracing: SpanContext not found in Extract carrier
```

This has also been seen on a cluster by @asubiotto

**Comment by dianasaur323** [comment]
@andreimatei I just marked this as 1.0, but feel free to move out if not mission critical.

**Comment by RaduBerinde** [comment]
Fixed by #15327.
